File: LaborMarket.Web/src/app/models/job-model.ts
import { EmployerDataModel } from './employer-model';

export type JobDataModel = {
  jobId: number;
  title: string;
  description: string;
  company: string;
  location: string;
  postedAt: Date;
  employerId: number;
  employer: EmployerDataModel;
};

export type CreateJobModel ={
  title: string;
  description: string;
  location: string;
  employerEmail: string;
}

File: LaborMarket.Api/Models/JobModels/JobModel.cs
using System.ComponentModel.DataAnnotations;

namespace LaborMarket.Api.Models
{
	public class JobModel
	{
		[Key]
		public int JobId { get; set; }
		public string Title { get; set; } = null!;
		public string Description { get; set; } = null!;
		public string Company { get; set; } = null!;
		public string Location { get; set; } = null!;
		public DateTime PostedAt { get; set; }
		public int EmployerId { get; set; }
		public EmployerModel Employer { get; set; } = null!;
	}
}
